Parsers/ASimAuthentication/Parsers/ASimAuthenticationCiscoMeraki.yaml

```diff
@@ -0,0 +1,155 @@
+      "12", "Other",
+      "13", "Logon violates policy",
+      "14", "Other",
+      "15", "Other",
+      "16", "Other",
+      "17", "Other",
+      "18", "Incorrect key",
+      "19", "Incorrect key",
+      "20", "Incorrect key",
+      "21", "Other",
+      "22", "Other",
+      "23", "Other",
+      "24", "Logon violates policy",
+  ];
+  let EventFieldsLookup = datatable (
+      LogSubtype: string,
+      EventResult: string,
+      EventType: string,
+      EventSeverity: string
+  )
+      [
+      "8021x_auth", "Success", "Logon", "Informational",
+      "wpa_auth", "Success", "Logon", "Informational",
+      "splash_auth", "Success", "Logon", "Informational",
+      "8021x_eap_success", "Success", "Logon", "Informational",
+      "8021x_deauth", "Success", "Logoff", "Informational",
+      "8021x_client_deauth", "Success", "Logoff", "Informational",
+      "wpa_deauth", "Success", "Logoff", "Informational",
+      "8021x_eap_failure", "Failure", "Logon", "Low",
+      "disassociation", "Failure", "Logon", "Low",
+  ];
+  let parser = (disabled: bool=false) {
+      union isfuzzy=true
+          (
+          meraki_CL
+          | project-rename LogMessage =  Message
+          ),
+          (
+          Syslog
+          | where Computer in (_ASIM_GetSourceBySourceType('CiscoMeraki'))
+          | project-rename LogMessage =  SyslogMessage
+          )
+      | where not(disabled)
+          and LogMessage has "events"
+          and (LogMessage has_any ("8021x_auth", "wpa_auth", "splash_auth", "8021x_deauth", "8021x_client_deauth", "wpa_deauth", "8021x_eap_failure", "8021x_eap_success") or (LogMessage has "disassociation" and LogMessage has "auth_neg_failed"))
+      | extend Parser = extract_all(@"(\d+.\d+)\s([\w\-\_]+)\s([\w\-\_]+)\s([\S\s]+)$", dynamic([1, 2, 3, 4]), LogMessage)[0]
+      | extend
+          Epoch = tostring(Parser[0]),
+          Device = tostring(Parser[1]),
+          LogType = tostring(Parser[2]),
+          Substring = tostring(Parser[3])
+      | where LogType == "events"
+      | parse Substring with * "type=" LogSubType:string " " restOfMessage:string
+      | where LogSubType in ("8021x_auth", "wpa_auth", "splash_auth", "8021x_deauth", "8021x_client_deauth", "wpa_deauth", "8021x_eap_failure", "8021x_eap_success") or (LogSubType == "disassociation" and Substring has "auth_neg_failed")
+      | extend EpochTimestamp = split(Epoch, ".")
+      | extend EventStartTime = unixtime_seconds_todatetime(tolong(EpochTimestamp[0]))
+      | extend EventEndTime = EventStartTime
+      | invoke _ASIM_ResolveDvcFQDN('Device')
+      | parse-kv Substring as(last_known_client_ip: string, ip: string, client_ip: string, client_mac: string, identity: string, reason: string, aid: string) with (pair_delimiter=" ", kv_delimiter="=", quote="'")
+      | extend Dvc = DvcHostname, 
+            aid = trim('"', aid)
+      | extend
+          DvcIpAddr = tostring(split(coalesce(last_known_client_ip, ip, client_ip), " ")[0]),
```

Comment:
If this is client IP, can we map it to SrcIpAddr?

Reply:
Done